### PR TITLE
[DowCrawler] Default to UTF-8 when possible

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #17258
| License       | MIT
| Doc PR        | -

This can't be ambiguous, let's use UTF-8 when possible.

### CODE REVIEW

**Comment by jakzal** [comment]
Looks good. Can we add a test please? :)

**Comment by jakzal** [comment]
Actually, would it ever [NOT match](https://3v4l.org/CWO6T) the regexp?

**Comment by nicolas-grekas** [comment]
yes: https://3v4l.org/EEQ9K

File: src/Symfony/Component/DomCrawler/Crawler.php [hunk 1]
```diff
@@ -127,8 +127,8 @@ public function add($node)
     /**
      * Adds HTML/XML content.
      *
-     * If the charset is not set via the content type, it is assumed
-     * to be ISO-8859-1, which is the default charset defined by the
+     * If the charset is not set via the content type, it is assumed to be UTF-8,
+     * or ISO-8859-1 as a fallback, which is the default charset defined by the
      * HTTP 1.1 specification.
```
Comment (fabpot) [inline src/Symfony/Component/DomCrawler/Crawler.php]: The end of the sentence does not make sense anymore.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**